rotkehlchen/chain/ethereum/modules/uniswap/utils.py

```diff
@@ -253,3 +273,319 @@ def historical_uniswap_v2_lp_price(
         token_price_func_args=[to_asset, timestamp],
         block_identifier=block_identifier,
     )
+
+
+def uniswap_v3_lp_token_balances(
+    userdb: 'DBHandler',
+    address: ChecksumEthAddress,
+    ethereum: 'EthereumManager',
+    premium: Optional[Premium],
+    known_assets: Set[EthereumToken],
+    unknown_assets: Set[EthereumToken],
+) -> List[NFTLiquidityPool]:
+    nft_manager_contract = EthereumContract(
+        address=UNISWAP_NFT_MANAGER_ADDRESS,
+        abi=UNISWAP_V3_NFT_MANAGER_ABI,
+        deployed_block=UNISWAP_V3_DEPLOYED_BLOCK,
+    )
+    address = to_checksum_address(address)
+    my_positions = nft_manager_contract.call(
+        ethereum=ethereum,
+        method_name="balanceOf",
+        arguments=[address],
+    )
+    balances: List[NFTLiquidityPool] = []
+    if my_positions <= 0:
+        return balances
+
+    chunks = list(get_chunks(list(range(my_positions)), n=UNISWAP_V3_POSITIONS_PER_CHUNK))
+    for chunk in chunks:
+        try:
```

Comment:
This is a huge for loop with 160++ lines of logic and without a single comment. I find it **extremely** hard to read and understand what it is you are doing here. Either break in smaller components or at least add some more comments explaining what is happening.